### ISSUE TITLE
add run function

### ISSUE THREAD

**Issue by coder966** [dannyvankooten/AltoRouter#235]
Hi. I would like to add a `run` method to start the router. This pattern (of defining the routes and then starting the router) typically exist in all the routers I know of, in Java, React, Node, etc...

This would make the library easier to start with and reduces the boilerplate code in users' code.

We can still let the user control what happens if no matching route found via passing an optional callable to the `run` method.

**Comment by coder966** [comment]
Any thoughts ?

**Comment by mxdpeep** [comment]
I use basically this:

`$alto = new \AltoRouter();
foreach ($presenter as $k => $v) {
    $alto->map($v["method"], $v["path"], $k, "route_${k}");
    if (substr($v["path"], -1) != "/") { // map slash endings
        $alto->map($v["method"], $v["path"] . "/", $k, "route_${k}_x");
    }
}`

**Comment by koenpunt** [comment]
Can you give an example of how that would work?

**Comment by koenpunt** [comment]
Oh I now see your PR

**Comment by mxdpeep** [comment]
> Oh I now see your PR

check `Tesseract LASAGNA` project `app.php`

**Comment by rootoor225** [comment]
Hello y'all !  Knowing hop usefull os altorouter that i discovered thanks to Grafikart on YouTube via his php lessons, i Would like to Know more With a good tutorial on YouTube

**Comment by rootoor225** [comment]
Tutorial about altorouter on YouTube will ne very usefull. The creators must think to this

**Comment by rootoor225** [comment]
Tutorial about altorouter on YouTube will ne very usefull. The creators must think to this

**Comment by ellisgl** [comment]
Yes, a person typing typos, humming, hawing and bumbling around in a youtube video is a great idea. Or even better, written text tutorial that can quickly fixed and quicker to get info.

**Comment by rootoor225** [comment]
Hello everybody ! 
How am you? I'm well but a little sad. 
Why?  Don't worry. Just because I've spent many times on YouTube to view videos about the best simple and useful routers I ever had used: AltoRouter. 
I sincerely have heard about it following Grafikart php tutorials videos. 
However, he didn't spend his time on explaining profoundly this router use. 
I exhort those of you guys, who are very best on teaching to make videos for teaching us, noobs, step by step, how to use altorouter. It will be very kind and good help.

Thanks and nice day to y'all.
